src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/StreamExtensions.cs

```diff
@@ -137,4 +140,24 @@ private unsafe static int ReadSize(Stream stream)
         return BitConverter.ToInt32(bytes, 0);
 #endif
     }
+
+#if NET
+    private static ValueTask ReadExactlyAsync(Stream stream, byte[] buffer, int length, CancellationToken cancellationToken)
+        => stream.ReadExactlyAsync(buffer, 0, length, cancellationToken);
+#else
+    private static async ValueTask ReadExactlyAsync(Stream stream, byte[] buffer, int length, CancellationToken cancellationToken)
```

Comment:
I'll do a follow up cleanup to move these to net only instead of having a bunch of ifdefs

Reply:
FWIW, none of these methods need to actually run on netstandard2.0 or NetFx. You could #ifdef the whole file for NET or change the file name to `StreamExtensions.NetCore.cs` so that they only compile for NET. Then, you can get rid of the non-NET branches. That will open up lots of Span-based APIs on Stream and other types.

Reply:
yea that's the plan, move the whole file over.